src/data/polynomial/degree.lean

```diff
@@ -148,6 +148,29 @@ end
 
 end injective
 
+section
+variable {f : polynomial R}
+
+lemma monomial_nat_degree_leading_coeff_eq_self (h : f.support.card ≤ 1) :
+  monomial f.nat_degree f.leading_coeff = f :=
```

Comment:
Could you please prove the following lemma? It may be useful in other cases too. Then in this lemma you can do `cases` + `simp`.
```
lemma finsupp.zero_or_single_of_card_support_le_one (f : α →₀ β) (h : f.support.card ≤ 1) :
  f = 0 ∨∃ a b, f = single a b :=
sorry
```

Reply:
I am not happy with the proof, but this works!

```
lemma finsupp.zero_or_single_of_card_support_le_one {α β : Type*} [has_zero β] (f : α →₀ β) (h : f.support.card ≤ 1) :
  f = 0 ∨∃ a b, f = single a b :=
begin
  by_cases s0 : f.support.card = 0,
  { left,
    rwa [card_eq_zero, support_eq_empty] at s0, },
  { right,
    have : ∃ a : α, f.support = singleton a,
    { rw ← card_eq_one,
      apply le_antisymm h _,
      rw [← ne.def, ← zero_lt_iff_ne_zero] at s0,
      exact nat.succ_le_iff.mp s0, },
    rcases this with ⟨a, ha⟩,
    use a,
    use f a,
    ext1,
    by_cases ha1 : a_1 = a,
    { subst ha1,
      rw single_eq_same, },
    { rw [(@not_mem_support_iff α β _inst_3 f a_1).mp, (@not_mem_support_iff α β _inst_3 _ a_1).mp],
      { intro,
        apply ha1,
        rw mem_support_single at a_2,
        exact a_2.1, },
      { rwa [ha, mem_singleton], }, }, },
end
```

Reply:
I wrote separate proofs for `card f.support = 0` and `card f.support = 1`.